tests/5.0/teams_loop/test_target_teams_loop_defaultmap.c

```diff
@@ -0,0 +1,238 @@
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(default)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = device_data[i];
+    device_data[i] = 2*i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+    OMPVV_TEST_AND_SET(errors, device_data[i] != 2*i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(firstprivate)
+  clause with target teams loop construct.
+*/
+int testDefaultMapFirstPrivate() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(firstprivate)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = device_data[i];
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != 0);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom:aggregate)
+  clause with target teams loop construct.
+*/
+int testDefaultMapToFromAggr() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(tofrom:aggregate)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = device_data[i];
+    device_data[i] = 2*i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+    OMPVV_TEST_AND_SET(errors, device_data[i] != 2*i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom:scalar)
+  clause with target teams loop construct.
+*/
+int testDefaultMapToFromScalar() {
+  int errors = 0;
+  int x = 0;
+#pragma omp target teams loop defaultmap(tofrom:scalar)
+  for (int i = 0; i < N; i++) {
+    x = N;
+  }
+  // checking results
+  OMPVV_TEST_AND_SET(errors, x != N);
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom:aggregate)
+  clause with target teams loop construct using structure.
+*/
+int testDefaultMapToFromAggrStr() {
+  int errors = 0;
+  struct dataS {
+    int device_data[N];
+    int host_data[N];
+  } dataObjS;
+  for (int i = 0; i < N; i++) {
+    dataObjS.device_data[i] = i;
+    dataObjS.host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(tofrom:aggregate)
+  for (int i = 0; i < N; i++) {
+    dataObjS.host_data[i] = dataObjS.device_data[i];
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, dataObjS.host_data[i] != i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom:pointer)
+  clause with target teams loop construct.
+*/
+int testDefaultMapToFromPtr() {
+  int errors = 0;
+  int *ptr = 0;
+#pragma omp target teams loop defaultmap(tofrom:pointer)
+  for (int i = 0; i < N; i++) {
+    ptr = (int*)0xefefefef;
```

Comment:
Multiple threads could be assigning to the same pointer variable. 
Either have an atomic write or an array of ptrs.

Reply:
Made changes as suggested...